### PR TITLE
EC-26: Implement Flyp.me. Views, CLI. Add docs and tests

### PR DESCRIPTION
**Pull request by dmytrostriletskyi**
Implemented functionality of `FLypMe`-exchanger.

**DONE**:
- [x] Views;
- [x] Add FlypMe to CLI and Rest endpoints;
- [x] Documentation.

Example of REST API usage:
```
curl -X PUT  -d '{ "from":"LTC", "to":"BTC", "amount": 1, "refund_address": "3CDJNfdWX8m2NwuGUV3nhXHXEeLygMXoAj", "destination": "1BoatSLRHtKNngkdXEeobR76b53LETtpyT" }'  
http://127.0.0.1:8000/api/v1/exchange/flypme/order/ | python -m json.tool
```

Example of CLI usage:
```
docker-compose -f docker-dev-compose.yml run essbase python cli.py
```

### CODE REVIEW

File: docs/source/config_ci_api.rst [hunk 1]
```diff
@@ -43,7 +43,8 @@ Payment
 ::
 
     docker-compose -f docker-dev-compose.yml run essbase python config.py payment:{tool-name} \
-        --set-defaults --data '{key}={value}, {key}={velue}'
+<<<<<<< HEAD
```
Comment (dmytrostriletskyi) [inline docs/source/config_ci_api.rst]: Forgot to clean merge conflicts points.

File: docs/source/rest_api.rst [hunk 2]
```diff
@@ -478,15 +478,292 @@ Swarm
...
+                "uuid": "fb7e2a23-a0f9-432d-865e-7f73c5cb6862"
+            }
+        }
+    }
+
+**Retrieve info about order.**
+
+::
+
+    curl 'http://127.0.0.1:8000/api/v1/exchange/flypme/order/?uuid=fb7e2a23-a0f9-432d-865e-7f73c5cb6862&status=True&info=True' \
```
Comment (dmytrostriletskyi) [inline docs/source/rest_api.rst]: Why did not you do the [same as here](https://github.com/essentiaone/ess-base/pull/21/files#diff-ac006f883fa45b3c0f36dcaa9512a10eR693)?

File: ess_base/ess_config/configs/networks/exchange/flypme.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "EXCHANGE_FLYPME_API_URL": "asfassfa"
+}
```
Comment (dmytrostriletskyi) [inline ess_base/ess_config/configs/networks/exchange/flypme.json]: Do you need this configs? Leave `{}` instead.

File: ess_base/ess_config/configs/networks/storage/swarm.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (dmytrostriletskyi) [inline ess_base/ess_config/configs/networks/storage/swarm.json]: Blank line, please.

File: ess_base/generic/cli/utils.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (dmytrostriletskyi) [inline ess_base/generic/cli/utils.py]: Remove blank line here.

File: ess_base/modules/exchange/flyp_me/flyp_me.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            refund_address=refund_address,
+            destination=destination,
+        )
+
+    @staticmethod
+    def update_order(uuid, amount, to_currency=None, from_currency=None, refund_address=None, destination=None):
+        """
+        Create order to exchange one currency to another.
+        """
+        print(destination)
```
Comment (dmytrostriletskyi) [inline ess_base/modules/exchange/flyp_me/flyp_me.py]: Remove print.

File: ess_base/modules/exchange/flyp_me/flyp_me.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    def get_order_info(uuid):
+        """
+        Return full info about order.
+        """
+        return modules.exchange.flyp_me.services.order.get_order_info(uuid=uuid)
+
+    @staticmethod
+    def get_exchange_limit(from_currency, to_currency):
+        """
+        Returns exchange limit for given currency-pair.
```
Comment (dmytrostriletskyi) [inline ess_base/modules/exchange/flyp_me/flyp_me.py]: Follow imperative mood, please. 
`Returns` -> `Return`.

File: ess_base/modules/exchange/flyp_me/flyp_me.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    def cancel_pending_order(uuid):
+        """
+        Cancel pending order.
+        """
+        return modules.exchange.flyp_me.services.order.cancel_pending_order(uuid=uuid)
+
+    @staticmethod
+    def check_order_status(uuid):
+        """
+        Returns status of order with given uuid.
```
Comment (dmytrostriletskyi) [inline ess_base/modules/exchange/flyp_me/flyp_me.py]: Follow imperative mood, please.
Returns -> Return.

File: ess_base/modules/exchange/flyp_me/services/order.py [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+        'order': {
+            'from_currency': from_currency,
+            'to_currency': to_currency,
+            'ordered_amount': amount,
+        },
+    }
+
+    if refund_address:
+        order['order']['refund_address'] = refund_address
+    if destination:
```
Comment (dmytrostriletskyi) [inline ess_base/modules/exchange/flyp_me/services/order.py]: Blank line above? For improving a visual indent.

File: docs/source/config_ci_api.rst (hunk 1 above)
Comment (a-ruban) [reply docs/source/config_ci_api.rst]: @dmytrostriletskyi done.

File: ess_base/ess_config/configs/networks/exchange/flypme.json (hunk 3 above)
Comment (a-ruban) [reply ess_base/ess_config/configs/networks/exchange/flypme.json]: @dmytrostriletskyi done.

File: ess_base/modules/exchange/flyp_me/flyp_me.py (hunk 6 above)
Comment (a-ruban) [reply ess_base/modules/exchange/flyp_me/flyp_me.py]: @dmytrostriletskyi done.

File: ess_base/modules/exchange/flyp_me/flyp_me.py (hunk 8 above)
Comment (a-ruban) [reply ess_base/modules/exchange/flyp_me/flyp_me.py]: @dmytrostriletskyi done.

File: ess_base/modules/exchange/flyp_me/flyp_me.py (hunk 7 above)
Comment (a-ruban) [reply ess_base/modules/exchange/flyp_me/flyp_me.py]: @dmytrostriletskyi done.

File: ess_base/modules/exchange/flyp_me/services/order.py (hunk 9 above)
Comment (a-ruban) [reply ess_base/modules/exchange/flyp_me/services/order.py]: @dmytrostriletskyi done.

File: docs/source/rest_api.rst (hunk 2 above)
Comment (a-ruban) [reply docs/source/rest_api.rst]: @dmytrostriletskyi fixed.

File: docs/source/base_cli_api.rst [hunk 10]
```diff
@@ -464,3 +464,237 @@ Swarm
...
+            "BC-GRC": "5.0013133208",
+            "BC-LTC": "0.0014622499",
+            "BC-PIVX": "0.0608649845",
+            "BC-PPC": "0.1218104665",
+            "BC-RBR": "28.8062765957",
+            "BC-SYS": "0.5795708553",
+            "BC-VTC": "0.0949427397",
+            "BC-ZEC": "0.0009787673",
+            "BC-ZRX": "0.3960468571",
+            .........................
```
Comment (dmytrostriletskyi) [inline docs/source/base_cli_api.rst]: Why many dots are using instead of common `...`?

File: docs/source/config_ci_api.rst [hunk 11]
```diff
@@ -34,16 +34,16 @@ Examples:
...
 --------
 
 Payment
 ~~~~~~~
 
 ::
 
-    docker-compose -f docker-dev-compose.yml run essbase python config.py payment:{tool-name} \
-        --set-defaults --data '{key}={value}, {key}={velue}'
+    docker-compose -f docker-dev-compose.yml run essbase python config.py payment:{tool-name} \<<
```
Comment (dmytrostriletskyi) [inline docs/source/config_ci_api.rst]: What is a `<<` here?

File: docs/source/rest_api.rst [hunk 12]
```diff
@@ -490,3 +490,281 @@ Swarm
...
+
+
+**Accept pending order.**
+
+::
+
+    curl -X POST http://127.0.0.1:8000/api/v1/exchange/flypme/pending_order/fb7e2a23-a0f9-432d-865e-7f73c5cb6862/ \
+    | python -m json.tool
+
+.. http:post::  /api/v1/exchange/flypme/pending_order/<uuid>
```
Comment (dmytrostriletskyi) [inline docs/source/rest_api.rst]: `/<uuid>` > `/<uuid>/`.

File: ess_base/adapters/exchange/views/flyp_me/limit.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    Provide API for work with Flyp.me exchange limit.
+    """
+
+    def get(self, request):  # pylint: disable=no-self-use
+        """
+        Retrieve exchange limits for given currencies.
+        """
+        currency_from = request.GET.get('from')
+        currency_to = request.GET.get('to')
+        try:
```
Comment (dmytrostriletskyi) [inline ess_base/adapters/exchange/views/flyp_me/limit.py]: Blank line as visual indent under `try`, please.

Comment (dmytrostriletskyi) [reply ess_base/adapters/exchange/views/flyp_me/limit.py]: The same as [this](https://github.com/essentiaone/ess-base/pull/21/files#diff-0066b6673fb543aeb9a9433f73c1c6e1R28), please. In all cases in code.

File: ess_base/modules/exchange/flyp_me/services/order.py [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+        },
+    }
+
+    if refund_address:
+        order['order']['refund_address'] = refund_address
+
+    if destination:
+        order['order']['destination'] = destination
+
+    order_data = json.dumps(order)
```
Comment (dmytrostriletskyi) [inline ess_base/modules/exchange/flyp_me/services/order.py]: `order` here is a `order data` too, `order_data` here is a `dumped_order_data`. Aren't they?

File: ess_base/providers/rest/api_settings/exchange/flyp_me.py [hunk 15]
```diff
@@ -0,0 +1,13 @@
+# pylint: disable=missing-docstring
```
Comment (dmytrostriletskyi) [inline ess_base/providers/rest/api_settings/exchange/flyp_me.py]: Easy to create documentation, no?

File: docs/source/config_ci_api.rst (hunk 11 above)
Comment (a-ruban) [reply docs/source/config_ci_api.rst]: @dmytrostriletskyi fixed.

File: docs/source/rest_api.rst (hunk 12 above)
Comment (a-ruban) [reply docs/source/rest_api.rst]: @dmytrostriletskyi fixed.

File: ess_base/adapters/exchange/views/flyp_me/rates.py [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    Provide API for work with Flyp.me exchange rates.
+    """
+
+    @staticmethod
+    def _get_rates_pair(request_get):  # pylint: disable=inconsistent-return-statements
+        """
+        Retrieve correct "from" and "to" params for request.
+        """
+        from_, to = request_get.get('from', None), request_get.get('to', None)  # pylint: disable=invalid-name
```
Comment (kozlov-oleksandr) [inline ess_base/adapters/exchange/views/flyp_me/rates.py]: @a-ruban , why **from_** with _ and **to** not?

File: ess_base/providers/rest/api_settings/exchange/flyp_me.py (hunk 15 above)
Comment (a-ruban) [reply ess_base/providers/rest/api_settings/exchange/flyp_me.py]: @dmytrostriletskyi fixed.

File: ess_base/modules/exchange/flyp_me/services/currency.py [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def get_active_currencies():
+    """
+    Return currencies.
+
+    A currency needs to have both exchange and send set to true to be enabled for the accountless exchange.
+
+    MissingHeaderBodySeparatorDefect is raised in background because of unparsed data - name of the currency.
+    So `0x` instead of normal name occurs.
```
Comment (kozlov-oleksandr) [inline ess_base/modules/exchange/flyp_me/services/currency.py]: @a-ruban , **0x** is not an error. See [here](https://coinmarketcap.com/currencies/0x/)

File: ess_base/modules/exchange/flyp_me/services/order.py (hunk 14 above)
Comment (a-ruban) [reply ess_base/modules/exchange/flyp_me/services/order.py]: @dmytrostriletskyi, fixed.

File: docs/source/base_cli_api.rst (hunk 10 above)
Comment (a-ruban) [reply docs/source/base_cli_api.rst]: @dmytrostriletskyi fixed.

File: ess_base/adapters/exchange/views/flyp_me/rates.py (hunk 16 above)
Comment (a-ruban) [reply ess_base/adapters/exchange/views/flyp_me/rates.py]: @kozlov-oleksandr  from - is a reserved word in Python.

**Pull request merged**